backend/src/main/java/gov/cdc/usds/simplereport/service/OrganizationService.java

```diff
@@ -476,28 +479,78 @@ public UUID getPermissibleOrgId(UUID orgId) {
     return orgId != null ? orgId : getCurrentOrganization().getInternalId();
   }
 
-  @AuthorizationConfiguration.RequireGlobalAdminUser
-  public List<UUID> getOrgAdminUserIds(UUID orgId) {
+  private List<ApiUser> getOrgAdminUsers(UUID orgId) {
     Organization org =
         organizationRepository.findById(orgId).orElseThrow(NonexistentOrgException::new);
-    List<String> adminUserEmails = oktaRepository.fetchAdminUserEmail(org);
-
-    return adminUserEmails.stream()
-        .map(
-            email -> {
-              Optional<ApiUser> foundUser = apiUserRepository.findByLoginEmail(email);
-              if (foundUser.isEmpty()) {
-                log.warn(
-                    "Query for admin users in organization "
-                        + orgId
-                        + " found a user in Okta but not in the database. Skipping...");
-              }
-              return foundUser.map(user -> user.getInternalId()).orElse(null);
-            })
-        .filter(Objects::nonNull)
+    try {
+      List<String> adminUserEmails = oktaRepository.fetchAdminUserEmail(org);
+      return adminUserEmails.stream()
+          .map(
+              adminUserEmail -> {
+                Optional<ApiUser> foundUser = apiUserRepository.findByLoginEmail(adminUserEmail);
+                if (foundUser.isEmpty()) {
+                  log.warn(
+                      "Query for admin users in organization "
+                          + org.getInternalId()
+                          + " found a user in Okta but not in the database. Skipping...");
+                }
+                return foundUser.orElse(null);
+              })
+          .filter(Objects::nonNull)
+          .collect(Collectors.toList());
+    } catch (IllegalGraphqlArgumentException e) {
```

Comment:
little confused as to when this catch would trigger. Would a GQL argument error not be related to the parameter passed in the method? In which case, could we just do an early return to save a level of error nesting here?

Help me understand if I'm missing something though! If somehow this get's triggered within the iteration for the user email stream, we might want to rename this particular error / reformat how we're doing the empty return to not need an extra catch.

Reply:
I ran into an issue while testing on dev4 with some org that existed in the DB but not in Okta which threw an exception here: https://github.com/CDCgov/prime-simplereport/blob/0be02bff31b2a159e0750dd61ba693a82e160c10/backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java#L560

I think what I will do is do a `try catch` here: https://github.com/CDCgov/prime-simplereport/blob/0be02bff31b2a159e0750dd61ba693a82e160c10/backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java#L593

something like
```
    try {
      List<User> admins = getOrgAdminUsers(org);
      return admins.stream().map(u -> u.getProfile().getLogin()).toList();
    } catch (IllegalGraphqlArgumentException e) {
      log.error(String.format("Error fetchAdminUserEmail:", e));
      return List.of();
    }
```

What do you think @fzhao99 ? 🤔

Reply:
Ahh I see, we're throwing a GQL error when there's an Okta-related issue with the list API. That was my main confusion but since we just throw a `IllegalGraphqlArgumentException` in the Okta repository don't think it makes sense to fix it here. 

I think we can just have the log statement be "error fetching admin by email from Okta API" or something similar since the stack trace would give us info as to what method would get called. Thanks for explaining!